lib/auth/grpcserver.go

```diff
@@ -4966,6 +4970,170 @@ func (g *GRPCServer) DeleteAllUserGroups(ctx context.Context, _ *emptypb.Empty)
 	return &emptypb.Empty{}, trace.Wrap(auth.DeleteAllUserGroups(ctx))
 }
 
+// ListOktaImportRules returns a paginated list of all Okta import rule resources.
+func (g *GRPCServer) ListOktaImportRules(ctx context.Context, req *oktapb.ListOktaImportRulesRequest) (*oktapb.ListOktaImportRulesResponse, error) {
```

Comment:
Did you consider implementing the RPCs in a separate struct/package? One of the benefits of having separate services is that we can start to break out from the AuthServer/GRPCServer monolithic packages.

A separate gRPC service is already loads better IMO, because it allows that refactor to occur, but it would be nice to go the extra mile if we can.

Reply:
I did, but the issue at the moment is `auth.Authorizer` is in the `auth` package. Basically there's a ton of stuff in `lib/auth` that needs to be extracted in order to properly set up a separate package like this.

Additionally, this will need to be backported to v12, which might not make this a good candidate for doing something like this.

Reply:
Yep, I keep hearing from various folks about the difficulties of dealing with some of the lib/auth dependencies. It would be nice to take the time and do whatever refactors need to be done. Still, I wish we could split this out - once it's in we are unlikely to backtrack.

WDYT about at least splitting into separate files, so we don't add to the already-gigantic grpcserver.go (and the likes).

Re backporting, I don't see an issue with backporting refactors; on the contrary, seems for the best.

Reply:
FWIW, here's a refactor of the authorizer that would make this easier: https://github.com/gravitational/teleport/pull/22776